modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteRequestBuilders.java

```diff
@@ -75,14 +75,13 @@ static Request initialSearch(SearchRequest searchRequest, BytesReference query,
             request.addParameter("scroll", keepAlive.getStringRep());
         }
         request.addParameter("size", Integer.toString(searchRequest.source().size()));
-        if (searchRequest.source().version() == null || searchRequest.source().version() == true) {
-            /*
-             * Passing `null` here just add the `version` request parameter
-             * without any value. This way of requesting the version works
-             * for all supported versions of Elasticsearch.
-             */
-            request.addParameter("version", null);
```

Comment:
I think we'll need the input from @nik9000 on this. In particular, it's unclear to me why we requested the version in case where the version was not set on `searchRequest`.

Reply:
It looks as if the version parameter always existed, with the default being false (the parameter was added in 0.15, see https://github.com/elastic/elasticsearch/issues/676). Also docs that go as far back as 0.90 seem to support this, see https://www.elastic.co/guide/en/elasticsearch/reference/0.90/search-request-version.html

I wonder if we should just write

```
request.addParameter("version", searchRequest.source().version());
```

here for simplicity. Would that break any tests?

Reply:
`addParameter` does not take a boolean. It takes a string. Additionally, searchRequest.source().version() returns `Boolean` the object so it can be null. I think that if we want to treat `null` as false (which I think we do as our modern concept of null I think means false, but an old version of Elasticsearch will be true) we need to explicitly handle each of null, false, and true.

Reply:
ok, makes sense.

Reply:
If I have read the comments correctly, I think this is a tiny breaking change, in that using remote reindex against some old version (0.9 or 0.90?) would previously request the version, making it available to scripts. I wonder if we need to add a breaking change documentation for it?
I find the new way reasonable and I also think this can be backported to 7.x, unless anyone has objections to this?

Reply:
> I find the new way reasonable and I also think this can be backported to 7.x, unless anyone has objections to this?

+1. I’m not aware of any scripts that would make use of the `version` field and reindex from 0.90 also becomes less and less common. I don't think there's a need for adding breaking-changes docs on this.